### PR TITLE
Robolectric and JaCoCo setup

### PR DESCRIPTION
**Pull request by mr-archano**
This set of changes in the `robolectric-tests/` module allows to enable code coverage reports using JaCoCo.
The main problem using the `java` plugin with Robolectric is that is ~~not easy~~ impossible to make JaCoCo picking the correct source folders to generate the reports. Applying the `robolectric` plugin along with the `com.android.library` one, sources are picked correctly instead.

**NOTE:** With this configuration it seems not possible to run tests in the IDE ([@blundell's solution](http://blog.blundell-apps.com/how-to-run-robolectric-junit-tests-in-android-studio/) does not work with this setup)

### CODE REVIEW

File: robolectric-tests/build.gradle [hunk 1]
```diff
@@ -1,32 +1,79 @@
-apply plugin: 'java'
+apply plugin: 'com.android.library'
```
Comment (blundell) [inline robolectric-tests/build.gradle:1]: strange, explain again why this change please?

File: robolectric-tests/build.gradle [hunk 2]
```diff
@@ -1,32 +1,79 @@
...
+        debug {
+            runProguard false
+        }
+        release {
+            runProguard false
+        }
+    }
+
+    lintOptions {
+        abortOnError false
```
Comment (blundell) [inline robolectric-tests/build.gradle:23]: tut

File: robolectric-tests/src/androidTest/java/com/malmstein/testingarchitecture/MyAndroidClassTest.java [hunk 3]
```diff
@@ -1,17 +1,15 @@
...
-import com.malmstein.testingarchitecture.RobolectricGradleTestRunner;
-
 import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.robolectric.annotation.Config;
 import org.robolectric.util.ActivityController;
 
 import static org.junit.Assert.assertEquals;
 
+@Config(emulateSdk = 18)
```
Comment (blundell) [inline robolectric-tests/src/androidTest/java/com/malmstein/testingarchitecture/MyAndroidClassTest.java:12]: vas is das?

File: robolectric-tests/build.gradle (hunk 1 above)
Comment (mr-archano) [inline robolectric-tests/build.gradle:1]: The explanation is in the PR description, but using the android library plugin you can run jacoco smoothly, using the correct set of sources.

File: robolectric-tests/build.gradle (hunk 2 above)
Comment (mr-archano) [inline robolectric-tests/build.gradle:23]: Default generated stuff. This should be true I guess

File: robolectric-tests/build.gradle (hunk 1 above)
Comment (blundell) [inline robolectric-tests/build.gradle:1]: I did read the description, was hoping for a bit more. I can't understand what the difference that matters is. Thats the only thing about plugins is you can't  _easily_ dive into them

File: robolectric-tests/src/androidTest/java/com/malmstein/testingarchitecture/MyAndroidClassTest.java (hunk 3 above)
Comment (mr-archano) [inline robolectric-tests/src/androidTest/java/com/malmstein/testingarchitecture/MyAndroidClassTest.java:12]: I am using robolectric 2.3, that actually needs `@Config` when targeting API > 18. Given this is a test project I can see the value of using the latest plugin, but if @malmstein doesn't like it I will change back to Robolectric 2.2

Comment (blundell) [inline robolectric-tests/src/androidTest/java/com/malmstein/testingarchitecture/MyAndroidClassTest.java:12]: ah this annotation is the same as having a custom test runner which changes the target sdk level :+1:

File: robolectric-tests/build.gradle (hunk 1 above)
Comment (mr-archano) [inline robolectric-tests/build.gradle:1]: The main problem is how the JaCoCo plugin behaves differently if used along with the `java` or `android-library` one. I could not find a solution for make it picking the sources from the android module using the `java` plugin. Crossing your solution with [this example](http://raptordigital.blogspot.co.uk/2014/08/code-coverage-reports-using-robolectric.html) I found a way to make it working.

File: robolectric-tests/src/androidTest/java/com/malmstein/testingarchitecture/MyAndroidClassTest.java (hunk 3 above)
Comment (mr-archano) [inline robolectric-tests/src/androidTest/java/com/malmstein/testingarchitecture/MyAndroidClassTest.java:12]: yep, but it supported only in robolectric 2.3+

**Comment by malmstein** [comment]
thanks @mr-archano :dancer:
